app/models/analyticEvent.server.ts

```diff
@@ -1,56 +1,187 @@
+  data: string;
+  count: number;
+};
+
+type AResByDay = {
+  event: string;
+  data: string;
+  count: number;
+  ts: string;
+};
 
 export async function getAnalyticsDataLast7Days(): Promise<Array<ARes>> {
   const startDate = new Date();
   startDate.setDate(startDate.getDate() - 7);
+  const startDateUnixTimestamp = Math.floor(startDate.getTime() / 1000);
 
   const analyticsData = await prisma.analyticEvent.groupBy({
     where: {
       ts: {
-        gte: startDate.toISOString(),
-      },
+        gte: startDateUnixTimestamp
+      }
     },
-    by: ['event', 'data', 'ip'],
+    by: ["event", "data", "ip"]
   });
 
-  const map = new Map<string,ARes>();
-  analyticsData.forEach(x => {
-    const key = x.event+x.data;
-    const prev = map.get(key) ?? {event:x.event, data:x.data, count:0};
-    prev.count +=1;
+  const map = new Map<string, ARes>();
+  analyticsData.forEach((x) => {
+    const key = x.event + x.data;
+    const prev = map.get(key) ?? { event: x.event, data: x.data, count: 0 };
+    prev.count += 1;
     map.set(key, prev);
   });
 
   var res = [...map.values()];
   res = res.sort((a, b) => b.count - a.count);
   return res;
-}
\ No newline at end of file
+}
+
+export async function tallyAnalytics(): Promise<Array<AResByDay>> {
+  try {
+    const latestTally = await prisma.analyticEventByDay.findFirst({
+      orderBy: {
+        day: "desc"
+      }
+    });
+
+    let startDate;
+    let endDate;
+
+    if (latestTally) {
+      endDate = new Date(latestTally.day);
+      startDate = new Date();
+      startDate.setDate(startDate.getDate() - 2);
+    } else {
+      startDate = new Date();
+      startDate.setDate(startDate.getDate() - 2);
+      endDate = new Date(0);
+    }
+    const startDateUnixTimestamp = Math.floor(startDate.getTime() / 1000);
+    const endDateUnixTimestamp = Math.floor(endDate.getTime() / 1000);
+
+    console.log("Start Date:", startDateUnixTimestamp);
+    console.log("End Date:", endDateUnixTimestamp);
+
+    const analyticsData = await prisma.analyticEvent.findMany({
+      where: {
+        ts: {
+          gt: endDateUnixTimestamp,
+          lt: startDateUnixTimestamp
+        }
+      }
+    });
+
+    const map = new Map<string, AResByDay>();
+    analyticsData.forEach((x) => {
+      const date = new Date(x.ts * 1000);
+      const year = date.getFullYear();
+      const month = String(date.getMonth() + 1).padStart(2, "0");
+      const day = String(date.getDate()).padStart(2, "0");
+
+      const formattedDate = `${year}-${month}-${day}`;
+      const key = x.event + x.data + formattedDate;
+      const prev = map.get(key) ?? {
+        event: x.event,
+        data: x.data,
+        count: 0,
+        ts: formattedDate
+      };
+      prev.count += 1;
+      map.set(key, prev);
+    });
+
+    var res = [...map.values()];
+    res = res.sort((a, b) => b.count - a.count);
+
+    for (const item of res) {
+      const existingRecord = await prisma.analyticEventByDay.findFirst({
```

Comment:
chatGPT can help you find the "upsert" method for prisma which will do this a little more conscisely

Reply:
Resolved.